ISSUE-223 - Directive execution on Type, Scalar, Enum, EnumValue. (#226)
dailymotion/tartiflette: docs/api/directive.md

* `on_build`: To wrap your directive around the build process of the SDL
* `on_field_execution`: Allows you to wrap the field execution. _(e.g. Access Rights on a specific field, apply a specific rate limit on a field.)_
* `on_argument_execution`: Allows you to wrap the argument execution. _(e.g Check the format of an input.)_.
* `on_pre_output_coercion`: Allows you to hook the execution flow right before a field result value is being coerced (Appears in version `0.10.0`)

> * `on_post_intput_coercien`: Allows you to hook the execution flow right after an argument value has beed coerced (Appears in version `0.10.0`)

Reviewer: ```suggestion
* `on_post_input_coercion`: Allows you to hook the execution flow right after an argument value has beed coerced (Appears in version `0.10.0`)
```

    @staticmethod
    async def on_post_input_coercion(
        directive_args: Dict[str, Any],
        next_directive: Callable,

>         val: Any,

Reviewer: ```suggestion
        value: Any,
```

        argument_definition: "GraphQLArgument",
        ctx: Optional[Dict[str, Any]],
        info: "Info",
    ) -> Any:
        ######################
        # Add your code here #
        ######################

>         return await next_directive(val, argument_definition, ctx, info)

Reviewer: ```suggestion
        return await next_directive(value, argument_definition, ctx, info)
```

    @staticmethod
    async def on_pre_output_coercion(
        directive_args: Dict[str, Any],
        next_directive: Callable,

>         val: Any,

Reviewer: ```suggestion
        value: Any,
```

        field_definition: "GraphQLField",
        ctx: Optional[Dict[str, Any]],
        info: "Info",
    ) -> Any:
        ######################
        # Add your code here #
        ######################

>         return await next_directive(val, field_definition, ctx, info)

Reviewer: ```suggestion
        return await next_directive(value, field_definition, ctx, info)
```
